master/buildbot/status/buildstep.py

```diff
@@ -380,27 +380,28 @@ def upgradeToVersion5(self):
         self.wasUpgraded = True
 
     @defer.inlineCallbacks
-    def prepare_trigger_links(self):
+    def prepare_trigger_links(self, codebases_arg):
         from buildbot.steps.trigger import Trigger
 
-        # for backward compatibility, the old pickled object does not have it.
-        if not hasattr(self, 'step_type_obj'):
-            return
-
         if not issubclass(self.step_type_obj, Trigger):
+            defer.returnValue([])
             return
 
+        links_data = []
         brids = self.build.brids
         db_results = yield self.build.builder.master.db.buildrequests\
                         .getBuildRequestForStartbrids(brids)
         master = self.build.builder.master
         for build in db_results:
             friendly_name = master.status.getFriendlyName(build['buildername'])
-            url = master.status.getURLForBuild(build['buildername'], build['number'],
+            url = master.status.getURLForBuild(build['buildername'],
```

Comment:
Is it possible to pass `codebases_arg` to the `getURLForBuild`  method? 
I think that since this method prepares URL already, we can add `codebase_arg` as nonrequired parameter and delegate to it the whole URL building process including GET parameters

Reply:
I thought about it but `getURLForBuild` has already `+getCodebasesArg(sourcestamps=sourcestamps)`
If I add next parameter `codebase_arg` I need to do something with this computation of `getCodebasesArg`. 
And then which parameter is more important? Can I use `sourcestamps` and `codebase_arg` together or not? Should I merge this 2 parameters before creating url params?

The simplest way was joining url with codebases_arg in prepare_trigger_links